design-tokens/build.js

```diff
@@ -1,93 +1,130 @@
 import { registerTransforms, transforms } from '@tokens-studio/sd-transforms';
-import * as fs from 'fs';
+import * as fs from 'node:fs';
 import StyleDictionary from 'style-dictionary';
 
-registerTransforms(StyleDictionary, {});
-
+const DEFAULT_THEME = 'light';
 const DESIGN_TOKENS_PATH = 'design-tokens/tokens';
 const BUILD_PATH = 'src/generated/';
 const NAME_PREFIX = 'design-tokens-';
 const BASE_TOKENS_NAME = `${NAME_PREFIX}base`;
 const CUSTOM_TRANSFORM_GROUP = 'sonar-design-tokens';
 const CUSTOM_FILTER_NO_COLOR = 'sonar-no-color';
-const CUSTOM_FILTER_NO_BASE = 'sonar-exclude-core';
+const CUSTOM_FILTER_THEMED_TOKENS = 'sonar-themed-tokens';
+const $themes = JSON.parse(fs.readFileSync(`${DESIGN_TOKENS_PATH}/$themes.json`, 'utf-8'));
 
-StyleDictionary.registerTransformGroup({
-  name: CUSTOM_TRANSFORM_GROUP,
-  transforms: ['attribute/cti', ...transforms, 'name/cti/kebab'],
-});
+initStyleDictionary();
+buildBaseTokens();
+buildThemedTokens($themes);
+buildCSSRootFile($themes);
+buildThemesEnumType($themes);
```

Comment:
You're redeclaring this `$themes` variable in the functions that use it (the argument), which might lead to weird bugs in the future.
How about creating an extra function to wrap that part?

```
function readAndBuildThemes() {
  const themes = parse(read())

  buildThemedTokens(themes);
  buildCSSRootFile(themes);
  buildThemesEnumType(themes);
}
```

NP: rename it `themeDefinitions`, to clarify what it is?

Reply:
ah yeah good point! 👍🏻

Reply:
I didn't add an extra function but renamed to something clearer and to avoid the naming collision